js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt

```diff
@@ -0,0 +1,210 @@
+package com.solanamobile.mobilewalletadapter.reactnative
+
+import android.net.Uri
+import android.os.Looper
+import android.util.Log
+import com.facebook.react.bridge.*
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.Semaphore
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.concurrent.ExecutionException
+
+class SolanaMobileWalletAdapterModule(reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext), CoroutineScope {
+
+    override val coroutineContext =
+        Dispatchers.IO + CoroutineName("SolanaMobileWalletAdapterModuleScope") + SupervisorJob()
+
+    companion object {
+        private const val ASSOCIATION_TIMEOUT_MS = 10000
+        private const val CLIENT_TIMEOUT_MS = 90000
+        private var localAssociation: LocalAssociationScenario? = null
+        private var mobileWalletAdapterClient: MobileWalletAdapterClient? = null
+        private var semTerminated: Semaphore? = null
+        private val mutex = Mutex()
+    }
+
+    override fun getName(): String {
+        return "SolanaMobileWalletAdapter"
+    }
+
+    @Throws(JSONException::class)
+    private fun convertMapToJson(readableMap: ReadableMap?): JSONObject {
+        val json = JSONObject()
+        val iterator = readableMap!!.keySetIterator()
```

Comment:
`!!` are super dangerous (basically the library/app would crash if it's really null).
There are a couple of them in this class, the best solution is usually to make the object non-null if possible, and/or to handle cases where it's null/empty for instance.

Reply:
Bang bang = Bad bad!

Reply:
The simplest refactor you can do here is:

```
return readableMap?.keySetIterator()?.let { iterator ->
    //TODO: This is where the rest of the code in the method goes
}
```

Reply:
Totally. Originally I had this as a `lateinit` which is a different kind of dangerous, but that didn't cut it since I wanted to _null them out_ in the cleanup – something that can't be done to `lateinit` vars.

Anyway, I literally don't understand why this is written this way, now that I look at it. I copy/pasted some Java code from somewhere and the IDE automatically converted it to Kotlin and did this to it. I didn't bother reading it over.